canopy_bioemi_mod.F90

```diff
@@ -0,0 +1,310 @@
+        REAL(RK) :: DTEMP_SHADE(SIZE(ZK))          ! Regression coefficient D for shade leaves
+        REAL(RK) :: TLEAF_SUN(SIZE(ZK))            ! Leaf temp for sunlit leaves (K)
+        REAL(RK) :: TLEAF_SHADE(SIZE(ZK))          ! Leaf temp for shaded leaves (K)
+        REAL(RK) :: TLEAF_AVE(SIZE(ZK))            ! Average Leaf temp (K)
+        REAL(RK) :: TLEAF24_AVE(SIZE(ZK))          ! Average Leaf temp over the past 24 hours (K)
+        REAL(RK) :: TLEAF240_AVE(SIZE(ZK))         ! Average Leaf temp over the past 240 hours (K)
+        REAL(RK) :: GammaTLEAF_SUN_NUM(SIZE(ZK))   ! Numerator in Tleaf sun activity factor
+        REAL(RK) :: GammaTLEAF_SHADE_NUM(SIZE(ZK)) ! Numerator in Tleaf shade activity factor
+        REAL(RK) :: GammaTLEAF_SUN_DEN(SIZE(ZK))   ! Denominator in Tleaf sun activity factor
+        REAL(RK) :: GammaTLEAF_SHADE_DEN(SIZE(ZK)) ! Denominator in Tleaf sun activity factor
+        REAL(RK) :: GammaTLEAF_SUN(SIZE(ZK))       ! Tleaf sun activity factor
+        REAL(RK) :: GammaTLEAF_SHADE(SIZE(ZK))     ! Tleaf shade activity factor
+        REAL(RK) :: GammaTLEAF_AVE(SIZE(ZK))       ! Average Tleaf activity factor
+        REAL(RK) :: PPFD24_SUN(SIZE(ZK))           ! Average PPFD sun over the past 24 hours
+        REAL(RK) :: PPFD24_SHADE(SIZE(ZK))         ! Average PPFD shade over the past 24 hours
+        REAL(RK) :: PPFD240_SUN(SIZE(ZK))          ! Average PPFD sun over the past 240 hours
+        REAL(RK) :: PPFD240_SHADE(SIZE(ZK))        ! Average PPFD shade over the past 240 hours
+        REAL(RK) :: CP_SUN(SIZE(ZK))               ! Normalized emission capacity sun at PPFD = 1000 umol phot/m2 s
+        REAL(RK) :: CP_SHADE(SIZE(ZK))             ! Normalized emission capacity shade at PPFD = 1000 umol phot/m2 s
+        REAL(RK) :: ALPHA_P_SUN(SIZE(ZK))          ! Quantum yield of isoprene sunlit (mol/mol)
+        REAL(RK) :: ALPHA_P_SHADE(SIZE(ZK))        ! Quantum yield of isoprene shade (mol/mol)
+        REAL(RK) :: GammaPPFD_SUN(SIZE(ZK))        ! PPFD activity factor sun (unitless)
+        REAL(RK) :: GammaPPFD_SHADE(SIZE(ZK))      ! PPFD activity factor shade (unitless)
+        REAL(RK) :: GammaPPFD_AVE(SIZE(ZK))        ! PPFD activity factor ave sun and shade
+        REAL(RK) :: E_OPT(SIZE(ZK))                ! maximum normalized emission capacity
+        REAL(RK) :: TLEAF_OPT(SIZE(ZK))            ! Tleaf at which E_OPT occurs (K)
+        REAL(RK) :: FLAI(SIZE(ZK))                 ! Fractional LAI in layer
+        REAL(RK) :: CT1                            ! Activation energy (kJ/mol)
+        REAL(RK) :: CEO                            ! Empirical coefficient
+        REAL(RK) :: EF                             ! Final Mapped Emission factor (EF) (ug/m2 hr)
+        integer i
+
+! Constant Canopy Parameters
+        REAL(RK),          PARAMETER     :: FRAC_PPFD       =  0.5_rk     !Fraction of incoming solar irradiance that is active PPFD
```

Comment:
"active PPFD"?? do you reference for this?

Reply:
Yes, the PAR incident at the top of the canopy is not equivalent to the total instantaneous downward shortwave flux (SWRAD) from the model, but can approximated as 50% the SWRAD.  I thought PAR was in the visible spectrum at about 400-700 nm, but the SWRAD is in shortwave spectrum from 300 - 4000 nm.

Reply:
Should call it `frac_par` instead.

Or another option would be to make PAR the input, the fraction already applied.

Reply:
@zmoon   IOK, [changed to FRAC_PAR](https://github.com/noaa-oar-arl/canopy-app/pull/50/commits/141d392bc2f9ae6bf5a203da5902c5975551e495).

Reply:
I think this is fine for now. We can come back to this at some point, maybe make the frac an input.